### PR TITLE
perf(create): run async tasks concurrently

### PR DESCRIPTION
**Pull request by sanjaiyan-dev**
Improving the performance by running the independent async functions in parallel ✈️.

_Sorry if I made any mistakes :(_

### CODE REVIEW

File: packages/docusaurus/package.json [hunk 1]
```diff
@@ -121,8 +121,8 @@
     "tree-node-cli": "^1.5.2"
   },
   "peerDependencies": {
-    "react": "^16.8.4 || ^17.0.0",
-    "react-dom": "^16.8.4 || ^17.0.0"
+    "react": "^16.8.4 || ^17.0.0 || ^18.0.0-0",
+    "react-dom": "^16.8.4 || ^17.0.0 || ^18.0.0-0"
```
Comment (Josh-Cena) [inline packages/docusaurus/package.json:125]: Please revert the unrelated changes.

File: packages/docusaurus/src/client/clientEntry.tsx [hunk 2]
```diff
@@ -5,47 +5,47 @@
...
-  });
-
-  // Webpack Hot Module Replacement API
-  if (module.hot) {
-    // Self-accepting method/ trick
-    // (https://github.com/webpack/webpack-dev-server/issues/100#issuecomment-290911036)
-    module.hot.accept();
-  }
-}
+ import React from 'react';
```
Comment (Josh-Cena) [inline packages/docusaurus/src/client/clientEntry.tsx:8]: Check the diff please, you still need to dedent

Comment (sanjaiyan-dev) [reply packages/docusaurus/src/client/clientEntry.tsx:8]: Extremely sorry i will change ASAP, sorry:(

File: packages/create-docusaurus/src/index.ts [hunk 3]
```diff
@@ -456,8 +456,10 @@ export default async function init(
   reqTemplate?: string,
   cliOptions: CLIOptions = {},
 ): Promise<void> {
-  const templates = await readTemplates();
-  const siteName = await getSiteName(reqName, rootDir);
+  const [templates, siteName] = await Promise.all([
+    readTemplates(),
+    getSiteName(reqName, rootDir)
```
Comment (Josh-Cena) [inline packages/create-docusaurus/src/index.ts:461]:
```suggestion
    getSiteName(reqName, rootDir),
```

**Review by Josh-Cena — APPROVED** [review]
Thanks
